model/utils.go

```diff
@@ -638,3 +638,82 @@ func GetPreferredTimezone(timezone StringMap) string {
 func IsSamlFile(saml *SamlSettings, filename string) bool {
 	return filename == *saml.PublicCertificateFile || filename == *saml.PrivateKeyFile || filename == *saml.IdpCertificateFile
 }
+
+// TN adding, delete useless comment
+func IsValidIp(ip string) bool {
+	return net.ParseIP(ip) != nil
+}
+func IsValidCidr(cidr string) bool {
+	_,_,err := net.ParseCIDR(cidr)
```

Comment:
I will put a space between.

Reply:
To solve all these minor mistakes and many others, you can ask the tool you are using to code to run gofmt for you. Will probably make things easier. Also, if I am not mistaken, you can run yourself gofmt and it will automatically solve all these.